File: cat_n_mouse.py
def cat_mouse(x,j):
    cat = x.find('C')
    dog = x.find('D')
    mouse = x.find('m')
    dist = abs(cat - mouse)
    if cat == -1:
        return 'boring without all three'
    elif dog == -1:
        return 'boring without all three'
    elif mouse == -1:
        return 'boring without all three'
    
    if dist > j:
        return 'Escaped!'
    elif dist <=j:
        if  (dog > cat and dog > mouse):
            return 'Caught!'
        elif (dog < cat and dog < mouse):
            return 'Caught!'
        elif (dog > cat and dog < mouse) or (dog< cat and dog > mouse):
            return 'Protected!'

File: catNMouse.js
function catMouse(x, j){
 
    let cat = x.indexOf('C');
  
    let mouse = x.indexOf('m');
  
    let dog = x.indexOf('D');
    
    let dist = Math.abs(cat - mouse);
    if (cat == -1 || mouse == -1 || dog == -1){
      return 'boring without all three';
    }
    if (dist > j){
      return 'Escaped!'
    }else if (dist <= j){
      if ( (dog > cat && dog > mouse) || (dog < cat && dog < mouse) ) {
         return 'Caught!';
      }else if ( (dog > cat && dog < mouse) ||  (dog < cat && dog > mouse) ){
         return 'Protected!'
      }
    } 
    }